File: src/libqpixmapcache.cpp
#include <QPixmap>
#include <QPixmapCache>
#define WORKAROUND_INNER_CLASS_DEFINITION_QPixmapCache__Key
#include <QString>
#include <QByteArray>
#include <cstring>
#include <qpixmapcache.h>
#include "libqpixmapcache.h"
#include "libqpixmapcache.hxx"

QPixmapCache* QPixmapCache_new(const QPixmapCache* other) {
    return new QPixmapCache(*other);
}

QPixmapCache* QPixmapCache_new2(QPixmapCache* other) {
    return new QPixmapCache(std::move(*other));
}

void QPixmapCache_CopyAssign(QPixmapCache* self, QPixmapCache* other) {
    *self = *other;
}

void QPixmapCache_MoveAssign(QPixmapCache* self, QPixmapCache* other) {
    *self = std::move(*other);
}

int QPixmapCache_CacheLimit() {
    return QPixmapCache::cacheLimit();
}

void QPixmapCache_SetCacheLimit(int cacheLimit) {
    QPixmapCache::setCacheLimit(static_cast<int>(cacheLimit));
}

bool QPixmapCache_Find(const libqt_string key, QPixmap* pixmap) {
    QString key_QString = QString::fromUtf8(key.data, key.len);
    return QPixmapCache::find(key_QString, pixmap);
}

bool QPixmapCache_Find2(const QPixmapCache__Key* key, QPixmap* pixmap) {
    return QPixmapCache::find(*key, pixmap);
}

bool QPixmapCache_Insert(const libqt_string key, const QPixmap* pixmap) {
    QString key_QString = QString::fromUtf8(key.data, key.len);
    return QPixmapCache::insert(key_QString, *pixmap);
}

QPixmapCache__Key* QPixmapCache_InsertWithPixmap(const QPixmap* pixmap) {
    return new QPixmapCache::Key(QPixmapCache::insert(*pixmap));
}

bool QPixmapCache_Replace(const QPixmapCache__Key* key, const QPixmap* pixmap) {
    return QPixmapCache::replace(*key, *pixmap);
}

void QPixmapCache_Remove(const libqt_string key) {
    QString key_QString = QString::fromUtf8(key.data, key.len);
    QPixmapCache::remove(key_QString);
}

void QPixmapCache_RemoveWithKey(const QPixmapCache__Key* key) {
    QPixmapCache::remove(*key);
}

void QPixmapCache_Clear() {
    QPixmapCache::clear();
}

void QPixmapCache_Delete(QPixmapCache* self) {
    delete self;
}

QPixmapCache__Key* QPixmapCache__Key_new() {
    return new QPixmapCache::Key();
}

QPixmapCache__Key* QPixmapCache__Key_new2(const QPixmapCache__Key* other) {
    return new QPixmapCache::Key(*other);
}

bool QPixmapCache__Key_OperatorEqual(const QPixmapCache__Key* self, const QPixmapCache__Key* key) {
    return (*self == *key);
}

bool QPixmapCache__Key_OperatorNotEqual(const QPixmapCache__Key* self, const QPixmapCache__Key* key) {
    return (*self != *key);
}

void QPixmapCache__Key_OperatorAssign(QPixmapCache__Key* self, const QPixmapCache__Key* other) {
    self->operator=(*other);
}

void QPixmapCache__Key_Swap(QPixmapCache__Key* self, QPixmapCache__Key* other) {
    self->swap(*other);
}

bool QPixmapCache__Key_IsValid(const QPixmapCache__Key* self) {
    return self->isValid();
}

void QPixmapCache__Key_Delete(QPixmapCache__Key* self) {
    delete self;
}

File: src/libqpixmapcache.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");

/// https://doc.qt.io/qt-6/qpixmapcache.html
pub const qpixmapcache = struct {
    /// New constructs a new QPixmapCache object.
    ///
    /// ``` other: QtC.QPixmapCache ```
    pub fn New(other: ?*anyopaque) QtC.QPixmapCache {
        return qtc.QPixmapCache_new(@ptrCast(other));
    }

    /// New2 constructs a new QPixmapCache object and invalidates the source QPixmapCache object.
    ///
    /// ``` other: QtC.QPixmapCache ```
    pub fn New2(other: ?*anyopaque) QtC.QPixmapCache {
        return qtc.QPixmapCache_new2(@ptrCast(other));
    }

    /// CopyAssign shallow copies `other` into `self`.
    ///
    /// ``` self: QtC.QPixmapCache, other: QtC.QPixmapCache ```
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QPixmapCache_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    ///
    /// ``` self: QtC.QPixmapCache, other: QtC.QPixmapCache ```
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QPixmapCache_MoveAssign(@ptrCast(self), @ptrCast(other));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qpixmapcache.html#cacheLimit)
    ///
    ///
    pub fn CacheLimit() i32 {
        return qtc.QPixmapCache_CacheLimit();
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qpixmapcache.html#setCacheLimit)
    ///
    /// ``` cacheLimit: i32 ```
    pub fn SetCacheLimit(cacheLimit: i32) void {
        qtc.QPixmapCache_SetCacheLimit(@intCast(cacheLimit));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qpixmapcache.html#find)
    ///
    /// ``` key: []const u8, pixmap: QtC.QPixmap ```
    pub fn Find(key: []const u8, pixmap: ?*anyopaque) bool {
        const key_str = qtc.struct_libqt_string{
            .len = key.len,
            .data = key.ptr,
        };
        return qtc.QPixmapCache_Find(key_str, @ptrCast(pixmap));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qpixmapcache.html#find)
    ///
    /// ``` key: QtC.QPixmapCache__Key, pixmap: QtC.QPixmap ```
    pub fn Find2(key: ?*anyopaque, pixmap: ?*anyopaque) bool {
        return qtc.QPixmapCache_Find2(@ptrCast(key), @ptrCast(pixmap));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qpixmapcache.html#insert)
    ///
    /// ``` key: []const u8, pixmap: QtC.QPixmap ```
    pub fn Insert(key: []const u8, pixmap: ?*anyopaque) bool {
        const key_str = qtc.struct_libqt_string{
            .len = key.len,
            .data = key.ptr,
        };
        return qtc.QPixmapCache_Insert(key_str, @ptrCast(pixmap));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qpixmapcache.html#insert)
    ///
    /// ``` pixmap: QtC.QPixmap ```
    pub fn Insert2(pixmap: ?*anyopaque) QtC.QPixmapCache__Key {
        return qtc.QPixmapCache_Insert2(@ptrCast(pixmap));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qpixmapcache.html#replace)
    ///
    /// ``` key: QtC.QPixmapCache__Key, pixmap: QtC.QPixmap ```
    pub fn Replace(key: ?*anyopaque, pixmap: ?*anyopaque) bool {
        return qtc.QPixmapCache_Replace(@ptrCast(key), @ptrCast(pixmap));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qpixmapcache.html#remove)
    ///
    /// ``` key: []const u8 ```
    pub fn Remove(key: []const u8) void {
        const key_str = qtc.struct_libqt_string{
            .len = key.len,
            .data = key.ptr,
        };
        qtc.QPixmapCache_Remove(key_str);
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qpixmapcache.html#remove)
    ///
    /// ``` key: QtC.QPixmapCache__Key ```
    pub fn Remove2(key: ?*anyopaque) void {
        qtc.QPixmapCache_Remove2(@ptrCast(key));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qpixmapcache.html#clear)
    ///
    ///
    pub fn Clear() void {
        qtc.QPixmapCache_Clear();
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qpixmapcache.html#dtor.QPixmapCache)
    ///
    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.QPixmapCache ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QPixmapCache_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qpixmapcache-key.html
pub const qpixmapcache__key = struct {
    /// New constructs a new QPixmapCache::Key object.
    ///
    ///
    pub fn New() QtC.QPixmapCache__Key {
        return qtc.QPixmapCache__Key_new();
    }

    /// New2 constructs a new QPixmapCache::Key object.
    ///
    /// ``` other: QtC.QPixmapCache__Key ```
    pub fn New2(other: ?*anyopaque) QtC.QPixmapCache__Key {
        return qtc.QPixmapCache__Key_new2(@ptrCast(other));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qpixmapcache__key.html#operator==)
    ///
    /// ``` self: QtC.QPixmapCache__Key, key: QtC.QPixmapCache__Key ```
    pub fn OperatorEqual(self: ?*anyopaque, key: ?*anyopaque) bool {
        return qtc.QPixmapCache__Key_OperatorEqual(@ptrCast(self), @ptrCast(key));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qpixmapcache__key.html#operator!=)
    ///
    /// ``` self: QtC.QPixmapCache__Key, key: QtC.QPixmapCache__Key ```
    pub fn OperatorNotEqual(self: ?*anyopaque, key: ?*anyopaque) bool {
        return qtc.QPixmapCache__Key_OperatorNotEqual(@ptrCast(self), @ptrCast(key));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qpixmapcache__key.html#operator=)
    ///
    /// ``` self: QtC.QPixmapCache__Key, other: QtC.QPixmapCache__Key ```
    pub fn OperatorAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QPixmapCache__Key_OperatorAssign(@ptrCast(self), @ptrCast(other));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qpixmapcache__key.html#swap)
    ///
    /// ``` self: QtC.QPixmapCache__Key, other: QtC.QPixmapCache__Key ```
    pub fn Swap(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QPixmapCache__Key_Swap(@ptrCast(self), @ptrCast(other));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qpixmapcache__key.html#isValid)
    ///
    /// ``` self: QtC.QPixmapCache__Key ```
    pub fn IsValid(self: ?*anyopaque) bool {
        return qtc.QPixmapCache__Key_IsValid(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.QPixmapCache__Key ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QPixmapCache__Key_Delete(@ptrCast(self));
    }
};
